faker/providers/person/en_NZ/__init__.py

```diff
@@ -0,0 +1,627 @@
+# coding=utf-8
+from __future__ import unicode_literals
+from collections import OrderedDict
+
+from .. import Provider as PersonProvider
+
+
+class Provider(PersonProvider):
+    formats = (
+        '{{first_name_male}} {{last_name}}',
+        '{{first_name_male}} {{last_name}}',
+        '{{first_name_male}} {{last_name}}',
+        '{{first_name_male}} {{last_name}}',
+        '{{first_name_male}} {{last_name}}-{{last_name}}',
+        '{{first_name_female}} {{last_name}}',
+        '{{first_name_female}} {{last_name}}',
+        '{{first_name_female}} {{last_name}}',
+        '{{first_name_female}} {{last_name}}',
+        '{{first_name_female}} {{last_name}}-{{last_name}}',
+    )
+
+    first_names_male = (
+        # Copied from en_GB
```

Comment:
I would rather compile a list from https://www.dia.govt.nz/diawebsite.nsf/wpg_URL/Services-Births-Deaths-and-Marriages-Most-Popular-Male-and-Female-First-Names?OpenDocument. Make sure you take English and Māori names from the same birth year, and cite the source URL in a comment

Reply:
First of all that would only be representative of people born in the last few years. Secondly, how do want to deal with the fact that New Zealanders don't have name suffixes? Setting `suffixes_male` and `suffixes_female` to empty tuples results in random.choice raising an IndexError.

Reply:
if you don't set `suffixes_*` at all, it should just work. the superclass' method (https://github.com/joke2k/faker/blob/master/faker/providers/person/__init__.py#L81) already checks if the attribute exists and returns an empty string if it doesn't.

Reply:
> that would only be representative of people born in the last few years

I think I'd be ok with that. If you can find a dataset for an earlier year that has both english and māori names, that'd would work too. I'd just rather use local data.